src/topology/test/backpressure.rs

```diff
@@ -0,0 +1,273 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
+use tokio::time::Duration;
+
+use crate::config::SinkOuter;
+
+use crate::topology::builder::PIPELINE_BUFFER_SIZE;
+use crate::{config::Config, test_util::start_topology};
+
+use vector_core::buffers::{BufferConfig, BufferType, WhenFull};
+
+pub const MEMORY_BUFFER_DEFAULT_MAX_EVENTS: usize = 500;
```

Comment:
Would it make sense to just call `memory_buffer_default_max_events` from `vector-core`? My concern is that they'll become out-of-sync. Granted this would just result in the test failing so it isn't a huge deal.